src/EditorFeatures/CSharpTest/Classification/SemanticClassifierTests.cs

```diff
@@ -2508,6 +2508,243 @@ interface unmanaged {}
+Regex.Text("-a"),
+Regex.CharacterClass("\\"),
+Regex.CharacterClass("w"),
+Regex.CharacterClass("\\"),
+Regex.CharacterClass("s"),
+Regex.Text("a"),
+Regex.CharacterClass("-"),
+Regex.Text("z"),
+Regex.CharacterClass("-"),
+Regex.CharacterClass("["),
+Regex.Text("m"),
+Regex.CharacterClass("-"),
+Regex.Text("p"),
+Regex.CharacterClass("]"),
+Regex.CharacterClass("]"),
+Regex.Quantifier("+"),
+Regex.Quantifier("?"),
+Regex.Comment("(?#comment)"),
+Regex.Alternation("|"),
+Regex.Grouping("("),
+Regex.Anchor("\\"),
+Regex.Anchor("b"),
+Regex.Anchor("\\"),
+Regex.Anchor("G"),
+Regex.Anchor("\\"),
+Regex.Anchor("z"),
+Regex.Grouping(")"),
+Regex.Alternation("|"),
+Regex.Grouping("("),
+Regex.Grouping("?"),
+Regex.Grouping("<"),
+Regex.Grouping("name"),
+Regex.Grouping(">"),
+Regex.Text("sub"),
+Regex.Grouping(")"),
+Regex.Quantifier("{"),
+Regex.Quantifier("0"),
+Regex.Quantifier(","),
+Regex.Quantifier("5"),
+Regex.Quantifier("}"),
+Regex.Quantifier("?"),
+Regex.Anchor("^"));
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task TestRegex2()
+        {
+            await TestAsync(
+@"
+using System.Text.RegularExpressions;
+
+class Program
+{
+
+    void Goo()
+    {
+        // language=regex
+        var r = @""$(\a\t\u0020)|[^\p{Lu}-a\w\sa-z-[m-p]]+?(?#comment)|(\b\G\z)|(?<name>sub){0,5}?^"";
+    }
+}",
+Keyword("var"),
+Regex.Anchor("$"),
+Regex.Grouping("("),
+Regex.OtherEscape("\\"),
+Regex.OtherEscape("a"),
+Regex.OtherEscape("\\"),
+Regex.OtherEscape("t"),
+Regex.OtherEscape("\\"),
+Regex.OtherEscape("u"),
+Regex.OtherEscape("0020"),
+Regex.Grouping(")"),
+Regex.Alternation("|"),
+Regex.CharacterClass("["),
+Regex.CharacterClass("^"),
+Regex.CharacterClass("\\"),
+Regex.CharacterClass("p"),
+Regex.CharacterClass("{"),
+Regex.CharacterClass("Lu"),
+Regex.CharacterClass("}"),
+Regex.Text("-a"),
+Regex.CharacterClass("\\"),
+Regex.CharacterClass("w"),
+Regex.CharacterClass("\\"),
+Regex.CharacterClass("s"),
+Regex.Text("a"),
+Regex.CharacterClass("-"),
+Regex.Text("z"),
+Regex.CharacterClass("-"),
+Regex.CharacterClass("["),
+Regex.Text("m"),
+Regex.CharacterClass("-"),
+Regex.Text("p"),
+Regex.CharacterClass("]"),
+Regex.CharacterClass("]"),
+Regex.Quantifier("+"),
+Regex.Quantifier("?"),
+Regex.Comment("(?#comment)"),
+Regex.Alternation("|"),
+Regex.Grouping("("),
+Regex.Anchor("\\"),
+Regex.Anchor("b"),
+Regex.Anchor("\\"),
+Regex.Anchor("G"),
+Regex.Anchor("\\"),
+Regex.Anchor("z"),
+Regex.Grouping(")"),
+Regex.Alternation("|"),
+Regex.Grouping("("),
+Regex.Grouping("?"),
+Regex.Grouping("<"),
+Regex.Grouping("name"),
+Regex.Grouping(">"),
+Regex.Text("sub"),
+Regex.Grouping(")"),
+Regex.Quantifier("{"),
+Regex.Quantifier("0"),
+Regex.Quantifier(","),
+Regex.Quantifier("5"),
+Regex.Quantifier("}"),
+Regex.Quantifier("?"),
+Regex.Anchor("^"));
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.Classification)]
+        public async Task TestRegex3()
```

Comment:
>TestRegex3 [](start = 26, length = 10)

Nit: it'd be nice to factor those three tests into one, since they share expectations.